### ISSUE TITLE
can't use a different port for ios

### ISSUE THREAD

**Issue by aprilmintacpineda** [facebook/react-native#33455]
### Description

I want to be able to start 2 metro bundlers together, to do that I needed to configure metro to run on different ports other than 8081, I set one app to run on `7000` and the other one to run on `7001` with the following steps:

1. Replace all `8081` with the desired port -- `7000` and `7001` -- there was only one result in the search, which was in `project.pbxproj`
2. run `react-native start --reset-cache --port 7000 --reset-cache`

Then I run android by running `cd android && ./gradlew clean && cd .. && react-native run-android --variant=devDebug --appIdSuffix=dev --port 7000` and it worked like a charm, on the other hand, I run ios through XCode and it seems that it can't connect to the desired port, I tried manually setting the port by shaking the device but even that doesn't work.

### Version

0.67.4

### Output of `npx react-native info`

```
System:
    OS: macOS 12.2.1
    CPU: (8) x64 Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz
    Memory: 1.89 GB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.0 - ~/.nvm/versions/node/v16.13.0/bin/node
    Yarn: 1.22.18 - ~/projects/samalapit/buyer/mobile-app/node_modules/.bin/yarn
    npm: 8.1.0 - ~/.nvm/versions/node/v16.13.0/bin/npm
    Watchman: 2021.12.06.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.2 - /Users/aprilmintacpineda/.rvm/gems/ruby-3.0.2/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.4, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK: Not Found
  IDEs:
    Android Studio: 2021.1 AI-211.7628.21.2111.8193401
    Xcode: 13.3/13E113 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.11 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.67.4 => 0.67.4 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

### Steps to reproduce

Included in the description above.

**Comment by aprilmintacpineda** [comment]
There's a solution state here but it did not work for me.
https://github.com/facebook/react-native/issues/30708#issuecomment-1007307841

**Comment by aprilmintacpineda** [comment]
I can confirm that it is still using 8081 as a port even after changing it to `7000` because when I closed the app, changed everything back to 8081, then start metro bundler (with 8081) and simply open the app (take note that I did not reinstall the app), it worked. So it's using 8081

**Comment by cortinico** [comment]
This was fixed in #33726
